src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -211,64 +211,62 @@ public override Task TrimCacheAsync()
 
             this.ScheduleCacheTrimmer(token =>
             {
-                string rootDirectory = Path.GetDirectoryName(this.CachedPath);
-
+                var rootDirectory = Path.GetDirectoryName(this.CachedPath);
                 if (rootDirectory != null)
                 {
-                    // Jump up to the parent branch to clean through the cache.
-                    // UNC folders can throw exceptions if the file doesn't exist.
-                    IEnumerable<string> directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse();
-
+                    // Jump up to the parent branch to clean through the cache
+                    // UNC folders can throw exceptions if the file doesn't exist
+                    var directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse().ToList();
```

Comment:
Even through `Reverse()` is using deferred execution, it still needs to enumerable all directories before it can return in reverse order. This makes sure it's executed on the same line and only does it once.